clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -761,7 +761,11 @@ public boolean refreshCommittedOffsetsIfNeeded(Timer timer) {
 
                 log.info("Setting offset for partition {} to the committed offset {}", tp, position);
                 entry.getValue().leaderEpoch().ifPresent(epoch -> this.metadata.updateLastSeenEpochIfNewer(entry.getKey(), epoch));
-                this.subscriptions.seekUnvalidated(tp, position);
+
+                // it's possible that the partition is no longer assigned when the response is received,
+                // so we need to ignore seeking if that's the case
+                if (this.subscriptions.isAssigned(tp))
```

Comment:
@hachikuji This is the change I make for auto offset reset.

I thought about adding a unit test but since the logic is wrapped in a single `refreshCommittedOffsetsIfNeeded` it is hard to change subscription in between without breaking it into multiple ones, but I feel it is too messy to test a singleton function so I did not add one.

Reply:
Does it make sense to move this check above a little bit?
```java
if (offsetAndMetadata != null && subscriptions.isAssigned(tp)) {
```
That will make the log message less confusing.

Reply:
Hmm, but I thought even if the partition is no longer assigned, we may still want to update its epoch; on the other hand your concern is valid. Will tweak it a bit more.